core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -431,28 +432,36 @@ public void remove() {
 
     @Override
     public void heartbeat() {
+      AtomicInteger removedSessions = new AtomicInteger(0);
       mActiveRegisterContexts.entrySet().removeIf((entry) -> {
         WorkerRegisterContext context = entry.getValue();
-        final long lastUpdate = mClock.millis() - context.getLastActivityTimeMs();
-        if (lastUpdate < mTimeout) {
+        final long clockTime = mClock.millis();
+        final long lastActivityTime = context.getLastActivityTimeMs();
+        final long staleTime = clockTime - lastActivityTime;
+        if (staleTime < mTimeout) {
           return false;
         }
         String msg = String.format(
-            "Worker register stream hanging for more than %sms for worker %d!"
+            "ClockTime: %d, LastActivityTime: %d. Worker %d register stream hanging for %sms!"
                 + " Tune up %s if this is undesired.",
-            mTimeout, context.mWorker.getId(),
-            PropertyKey.MASTER_WORKER_REGISTER_STREAM_RESPONSE_TIMEOUT.toString());
+            clockTime, lastActivityTime, context.mWorker.getId(), staleTime,
+            PropertyKey.MASTER_WORKER_REGISTER_STREAM_RESPONSE_TIMEOUT);
```

Comment:
slightly improved the error msg so it's more obvious when the stale time looks fishy